text/0000-immovable-types.md

```diff
@@ -119,6 +119,17 @@ I suggest we do a `crater` run to investigate if these breakages are feasible.
 
 Changing these associated types will be insta-stable. You would be unable to write stable code which would conflict with this proposal. `?Move` bounds would also show up in documentation, although we would be able to filter those out if desired.
 
+## Allowing immovable types in container types`Box`
+
+`std::boxed::Box`, `std::rc::Rc`, `std::rc::Weak`, `std::sync::Arc`, `std::sync::Weak` will be changed to allow immovable types inside, but will themselves be movable. These can be used to overcome the limitations of immovable types at the cost of an allocation.
+
+For `Rc` and `Arc` , the function `try_unwrap` would only be allowed on movable types.
+
+In general, we can allow immovable types in an movable container if we either:
+- disallow all methods of accessing the address of the contained immovable types, including references (possible for `Vec`, `HashMap`)
```

Comment:
That seems confusing, since you can get the address of both l and r-values in Rust. Not that these are important enough to have names.